src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts

```diff
@@ -0,0 +1,284 @@
+
+import {
+  SkySummaryActionBarTestingModule
+} from './summary-action-bar.module';
+
+describe('Summary action bar fixture', () => {
+  let fixture: ComponentFixture<SummaryActionBarTestComponent>;
+  let testComponent: SummaryActionBarTestComponent;
+  let mockQueryService: MockSkyMediaQueryService;
+  let summaryActionBarFixture: SkySummaryActionBarFixture;
+
+  //#region helpers
+
+  async function initiateResponsiveMode(breakpoint: SkyMediaBreakpoints): Promise<void> {
+    mockQueryService.fire(breakpoint);
+    fixture.detectChanges();
+    return fixture.whenStable();
+  }
+
+  //#endregion
+
+  beforeEach(() => {
+    // replace the mock service before using in the test bed to avoid change detection errors
+    mockQueryService = new MockSkyMediaQueryService();
+
+    TestBed.configureTestingModule({
+      declarations: [
+        SummaryActionBarTestComponent
+      ],
+      imports: [
+        SkyKeyInfoModule,
+        SkySummaryActionBarTestingModule
+      ],
+      providers: [
+        { provide: SkyMediaQueryService, useValue: mockQueryService }
+      ]
+    });
+
+    fixture = TestBed.createComponent(
+      SummaryActionBarTestComponent
+    );
+    testComponent = fixture.componentInstance;
+    fixture.detectChanges();
+    summaryActionBarFixture = new SkySummaryActionBarFixture(
+      fixture,
+      SummaryActionBarTestComponent.dataSkyId
+    );
+  });
+
+  describe('primary action', () => {
+
+    it('should expose primary action button default properties', async () => {
+      const actionSpy = spyOn(fixture.componentInstance, 'primaryActionClicked');
+      const action = summaryActionBarFixture.primaryAction;
+
+      // verify button state
+      expect(action.buttonText).toEqual(testComponent.primaryAction.buttonText);
+      expect(action.isDisabled).toEqual(testComponent.primaryAction.isDisabled);
+
+      // verify the click methods work on the button state
+      await action.click();
+      expect(actionSpy).toHaveBeenCalled();
+    });
+
+    it('should expose primary action button default properties when responsive', async () => {
+      await initiateResponsiveMode(SkyMediaBreakpoints.xs);
+      const actionSpy = spyOn(fixture.componentInstance, 'primaryActionClicked');
+      const action = summaryActionBarFixture.primaryAction;
+
+      // verify button state
+      expect(action.buttonText).toEqual(testComponent.primaryAction.buttonText);
+      expect(action.isDisabled).toEqual(testComponent.primaryAction.isDisabled);
+
+      // verify the click methods work on the button state
+      await action.click();
+      expect(actionSpy).toHaveBeenCalled();
+    });
+
+    it('should expose primary action button properties', async () => {
+      // modify the defaults
+      testComponent.primaryAction = {
+        buttonText: 'some action',
+        isDisabled: true,
+        click: () => testComponent.primaryActionClicked()
+      };
+      fixture.detectChanges();
+      await fixture.whenStable();
```

Comment:
Is this line needed? This test might be able to not be `async`? I could be wrong - just wanted to ask.

Reply:
I'm sure you could get away with not waiting for the fixture to stabilize, but I'd rather avoid introducing any timing issues based on an aversion to using async. This seems more explicit to me; wait for the changes to take place before proceeding.

Reply:
I get what you're saying. I think here though we are just changing some values so there really isn't anything to stabilize as changing those values does not start any async operations. Angular's change detection will happen via the `detectChanges` call and we are just changing bindings and so there isn't anything to stabilize. I also think that with this being in the `spec` that we don't need to worry about what might happen for others - if it works without it then it works. We aren't having to deal with any consumer code that might be wonky in this case like we might be dealing with in the fixture itself.

Reply:
Sure